dependencies/CMakeLists.txt

```diff
@@ -22,11 +22,17 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 # SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-include(FetchContent)
 
-FetchContent_Declare(date
-    GIT_REPOSITORY https://github.com/HowardHinnant/date.git
-    GIT_TAG        v3.0.0
-) 
+if(USE_SYSTEM_DATE)
+    find_package(Threads REQUIRED)
```

Comment:
Why Threads here? 

If we need this, then presumbably only on **non** msvc and you would also need to link against it. But I'm not sure if we do. cc: @rbock 

It might be that wee need this and relied on Date for this https://github.com/HowardHinnant/date/blob/master/CMakeLists.txt#L152. 
While this should work fine in both modes, it would be better to do this here as well if we need it to not rely on date providing it. But in these case we should find_package(Threads) in both cases I assume (and link against it)

Reply:
I've added that in the first PR revision because it did not compile on my system without that. Now I've checked that it works correctly without it (and it is expected - date library marks Threads as a dependency, and it is found automatically: https://github.com/HowardHinnant/date/blob/master/cmake/dateConfig.cmake#L4). Deleting this line (as not needed).